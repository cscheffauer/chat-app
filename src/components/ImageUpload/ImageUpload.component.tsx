import React, { useRef, SyntheticEvent } from 'react';
import './ImageUpload.scss';

interface Props {
	openImagePreview: (e: SyntheticEvent<HTMLInputElement>) => void;
}

const ImageUpload = ({ openImagePreview }: Props) => {
	const fileUploadRef = useRef<HTMLInputElement>(null);

	return (
		<>
			<label className='imageupload' htmlFor='imageupload'>
				&#128248;
			</label>
			<input id='imageupload' accept='image/png, image/jpeg' multiple={false} onChange={openImagePreview} type='file' ref={fileUploadRef} style={{ display: 'none' }} />
		</>
	);
};

export default ImageUpload;
