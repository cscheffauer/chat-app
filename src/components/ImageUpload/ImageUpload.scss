.imageupload {
	position: absolute;
	left: 5px;
	top: 15%;
	height: 75%;
	width: 40px;
	font-size: 3vh;
	line-height: 100%;
	text-align: center;
	cursor: pointer;
}
