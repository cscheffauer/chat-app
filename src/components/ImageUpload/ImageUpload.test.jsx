import React from 'react';
import { mount } from 'enzyme';
import ImageUpload from './ImageUpload.component';

const username = 'Max Mustermann';

describe('<ImageUpload />', () => {
	let wrapper;
	const mockOpenImagePreview = jest.fn();

	beforeEach(() => {
		wrapper = mount(<ImageUpload openImagePreview={mockOpenImagePreview} />);
	});

	afterEach(() => {
		jest.clearAllMocks();
	});

	it('expect to render ImageUpload component', () => {
		expect(wrapper).toMatchSnapshot();
	});

	describe('Testing the ImageUpload', () => {
		it('expect to have the function called on a input value change', () => {
			wrapper
				.find('input')
				.props()
				.onChange({ preventDefault() {} }); //change input
			expect(mockOpenImagePreview).toHaveBeenCalled();
		});
	});
});
