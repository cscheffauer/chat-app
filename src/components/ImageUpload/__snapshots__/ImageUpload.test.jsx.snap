// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageUpload /> expect to render ImageUpload component 1`] = `
<ImageUpload
  openImagePreview={[MockFunction]}
>
  <label
    className="imageupload"
    htmlFor="imageupload"
  >
    ðŸ“¸
  </label>
  <input
    accept="image/png, image/jpeg"
    id="imageupload"
    multiple={false}
    onChange={[MockFunction]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
</ImageUpload>
`;
